src/AWS.Deploy.Common/Exceptions.cs

```diff
@@ -318,5 +318,26 @@ public static string PrettyPrint(this Exception? e)
 
             return $"{Environment.NewLine}{e.Message}{Environment.NewLine}{e.StackTrace}{PrettyPrint(e.InnerException)}";
         }
+
+        /// <summary>
+        /// Returns the truncated error message by only preserving the leading and trailing k characters.
+        /// The message will only be truncated if its length is greater than 2*k.
+        /// This method returns <see cref="string.Empty"/> if the error message is null or empty.
+        /// </summary>
+        public static string GetTruncatedErrorMessage(this Exception? e, int k = 500)
+        {
+            var message = e?.Message;
+
+            if (string.IsNullOrEmpty(message))
+                return string.Empty;
+
+            if (message.Length <= 2*k)
+                return message;
+
+            var firstKChars = message.Substring(0, k);
+            var lastkChars = message.Substring(message.Length - k);
+            var seperator = $"{Environment.NewLine}...{Environment.NewLine}...{Environment.NewLine}...{Environment.NewLine}";
+            return $"This error message is truncated to only include leading and trailing {k} characters:{Environment.NewLine}{firstKChars}{seperator}{lastkChars}";
```

Comment:
* You're truncating to 1000 characters by default. but you append 83 characters (close to 10% of the length) onto it. Can this be shortened a little?
* Would it make sense to put this in the truncation instead of at the start of the error? Something like
```
error text
error text
...
Error truncated to the first and last 500 characters
...
error text
```

Reply:
Changed the truncated message to match the suggestion.